### PR TITLE
Push image to ghcr.io

### PR DESCRIPTION
**Pull request by Roming22**
Images are still being pushed to quay.io

Signed-off-by: Romain Arnaud <rarnaud@redhat.com>

### CODE REVIEW

File: .github/workflows/shellcheck-image.yaml [hunk 1]
```diff
@@ -39,5 +39,14 @@ jobs:
           registry: quay.io/redhat-pipeline-service
           username: ${{ secrets.QUAY_USERNAME }}
           password: ${{ secrets.QUAY_TOKEN }}
+      - name: Push to ghcr.io
+        id: push-to-ghcr
+        uses: redhat-actions/push-to-registry@v2
+        with:
+          image: ${{ steps.build-image.outputs.image }}
+          tags: ${{ steps.build-image.outputs.tags }} ${{ github.ref_name }}
+          registry: ghcr.io/${{ github.repository_owner }}
```
Comment (xinredhat) [inline .github/workflows/shellcheck-image.yaml:48]: shellcheck image is for testing, shall we expect those images to be under something like test/images?

Comment (Roming22) [reply .github/workflows/shellcheck-image.yaml]: ghcr is an image registry so `images` seems a bit redundant, but I've added `test` to the path.

File: .github/workflows/shellcheck-image.yaml [hunk 2]
```diff
@@ -39,5 +39,14 @@ jobs:
           registry: quay.io/redhat-pipeline-service
           username: ${{ secrets.QUAY_USERNAME }}
           password: ${{ secrets.QUAY_TOKEN }}
+      - name: Push to ghcr.io
+        id: push-to-ghcr
+        uses: redhat-actions/push-to-registry@v2
+        with:
+          image: ${{ steps.build-image.outputs.image }}
+          tags: ${{ steps.build-image.outputs.tags }} ${{ github.ref_name }}
+          registry: ghcr.io/${{ github.repository_owner }}/test
```
Comment (fgiloux) [inline .github/workflows/shellcheck-image.yaml:48]: Having  the images in different paths in different registries, defeat some of the purpose of having them in multiple registries. In the past using sub-paths for storing images was not supported. I am not sure if that is still the case with some of the registries. If it is I see two options:
- we say that we don't ship test images and as such we store them only in ghcr.io
- we create a second organization: ${{ github.repository_owner }}-test or something similar

In this respect having different names for the organizations in quay.io and ghcr.io defeats the purpose of having the images in multiple registries.

**Review by fgiloux — CHANGES_REQUESTED** [review]

File: .github/workflows/shellcheck-image.yaml (hunk 2 above)
Comment (Roming22) [reply .github/workflows/shellcheck-image.yaml:48]: How about I prefix test images with `test-` and leave them at the root of the registry?

Comment (Roming22) [reply .github/workflows/shellcheck-image.yaml]: I implemented the change, let me know if you're not happy with the solution.

Comment (fgiloux) [reply .github/workflows/shellcheck-image.yaml]: I understand that it is not directly related to this PR but I am still concerned with 
ghcr.io/openshift-pipelines vs quay.io/redhat-pipeline-service
If users do `docker pull openshift-pipelines/cluster-setup:latest` and they don't have ghcr.io configured in their registry list best case they don't get the image, worst case they get someone's else image, possibly from an attacker.

Comment (fgiloux) [reply .github/workflows/shellcheck-image.yaml]: I thought that it is what we wanted to address by publishing to multiple registries.

Comment (bnallapeta) [reply .github/workflows/shellcheck-image.yaml]: +1 to @fgiloux 's comment. I created this #174 a while back to track this.

Comment (Roming22) [reply .github/workflows/shellcheck-image.yaml]: I also agree on the need to push to docker. That would be a separate PR though.

**Pull request closed without merging**